src/s3fetchers.ts

```diff
@@ -0,0 +1,113 @@
+import { Readable } from 'stream';
+
+import { S3Client, ListObjectsV2Command, GetObjectCommand } from "@aws-sdk/client-s3";
+
+import { BlockHeight, LakeConfig, Block, Shard, StreamerMessage } from './types';
+import { prependZeroes, parseBody } from './utils';
+
+// Queries the list of the objects in the bucket, grouped by "/" delimiter.
+// Returns the list of blocks that can be fetched
+export async function listBlocks(
+  client: S3Client,
+  bucketName: string,
+  startAfter: BlockHeight,
+): Promise<BlockHeight[]> {
+  try {
+    const data = await client.send(
+        new ListObjectsV2Command({
+          Bucket: bucketName,
+          MaxKeys: 10,
+          Delimiter: "/",
+          StartAfter: prependZeroes(startAfter),
+          RequestPayer: "requester",
+        })
+    );
+    const blockHeights = data.CommonPrefixes.map(p => parseInt(p.Prefix.split('/')[0]));
```

Comment:
Is there any context I can read about? How does S3 format look like?

Reply:
Not sure if it helps https://docs.aws.amazon.com/AmazonS3/latest/API/API_ListObjectsV2.html

Reply:
And if you were asking about OUR data structure you can read about it here https://github.com/near/near-lake#data-structure

Reply:
@khorolets Let's leave this comment right there in the code.